### PR TITLE
feat: support paging to get a list of objects 

### LINKED ISSUE #7320: feat: new API design for the Admin API
### Description

One of the plans in the APISIX V3 roadmap was discussed in the mailing list: new API design for the Admin API

mail list link: https://lists.apache.org/thread/jg46m1lxn2zx3gz1r77b3v6cz6v2l4bn

Now I'm going to implement it, and this task is split into the following subtasks:

- [ ] updated some list URIs(SSL, proto )
- [ ] response body format
- [ ] supports paging to get a list of objects such as route
- [ ] supports filter when getting a list of objects such as route
- [ ] additional filter fields supported by the Route object

This issue is used to track the progress of this task, and also used to record things that require collaboration with other projects (dashboard, apisix-ingress-controller)

### PR DESCRIPTION
**Pull request by tzssangglass**
### Description

Fixes #7320

wait for: #7366

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: apisix/admin/v3_adapter.lua [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    end
+end
+
+
+local function sort(l, r)
+    return l.createdIndex < r.createdIndex
+end
+
+
+function _M.pagination(body)
```
Comment (spacewander) [inline apisix/admin/v3_adapter.lua:84]: Can we call it `filter`? So the new feature can be added in this method, without modifying lots of admin resource modules.

File: apisix/admin/v3_adapter.lua [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    if args.page_size < 10 or args.page_size > 500 then
+        return response.exit(400, "page_size must be between 10 and 500")
+    end
+
+    if not args.page or args.page < 1 then
+        -- default page is 1
+        args.page = 1
+    end
+
+    local list = (body.node and body.node.nodes) or body.list
```
Comment (spacewander) [inline apisix/admin/v3_adapter.lua:106]: body.node.nodes is nil in v3

File: apisix/admin/v3_adapter.lua [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+    -- sort nodes by there createdIndex
+    table.sort(list, sort)
+
+    local to = args.page * args.page_size
+    local form =  to - args.page_size + 1
```
Comment (spacewander) [inline apisix/admin/v3_adapter.lua:112]:
```suggestion
    local from =  to - args.page_size + 1
```

File: apisix/admin/v3_adapter.lua [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+    local res = table.new(args.page_size, 0)
```
Comment (spacewander) [inline apisix/admin/v3_adapter.lua:113]: It seems people can pass a big page_size (> len(list)) and force the Admin API to create a big table?

File: apisix/admin/v3_adapter.lua [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+    if not enable_v3() then
+        return
+    end
+
+    local args = request.get_uri_args()
+    if not args.page or not args.page_size then
+        return
+    end
+
+    args.page = tonumber(args.page)
```
[the remark this answers is not in the document]
Comment (tzssangglass) [reply apisix/admin/v3_adapter.lua]: updated

File: apisix/admin/v3_adapter.lua (hunk 1 above)
Comment (tzssangglass) [reply apisix/admin/v3_adapter.lua]: renamed it

File: apisix/admin/v3_adapter.lua (hunk 2 above)
Comment (tzssangglass) [reply apisix/admin/v3_adapter.lua]: updated

File: apisix/admin/v3_adapter.lua (hunk 4 above)
Comment (tzssangglass) [reply apisix/admin/v3_adapter.lua]: use 20 as default value

**Review by soulbird — APPROVED** [review]

**Review by spacewander — APPROVED** [review]

**Pull request merged**